pkg/proxy/topology.go

```diff
@@ -93,6 +93,10 @@ func CategorizeEndpoints(endpoints []Endpoint, svcInfo ServicePort, nodeLabels m
 				hasLocalServingTerminatingEndpoints = true
 			}
 		}
+		
+		if hasLocalReadyEndpoints && hasLocalServingTerminatingEndpoints {
```

Comment:
Based on https://github.com/kubernetes/kubernetes/pull/119532#issuecomment-1660446767,  i think the change still makes sense, it does not affect the result of hasLocalServingTerminatingEndpoints, even though the improvement it brings may be marginal.
WDYT @danwinship

Reply:
great catch @HirazawaUi